app/workers/historical_data_deletion_worker.rb

```diff
@@ -0,0 +1,81 @@
+class HistoricalDataDeletionWorker < ApplicationWorker
+  def perform
+    # cascades matched content changes
+    delete_and_log(ContentChange) { ContentChange.where("created_at < ?", retention_period) }
+
+    # cascades matched messages
+    delete_and_log(Message) { Message.where("created_at < ?", retention_period) }
+
+    # cascades digest run subscribers
+    delete_and_log(DigestRun) { DigestRun.where("created_at < ?", retention_period) }
+
+    # allows subscriber lists and subscribers to be deleted
+    delete_and_log(Subscription) { Subscription.where("ended_at < ?", retention_period) }
+
+    # cascades matched content changes and matched messages
```

Comment:
I guess we don't expect this to do any cascading as these would already be dealt with?

Reply:
I've removed the comment 👍